src/storage/memtable/ob_memtable.cpp

```diff
@@ -848,9 +848,37 @@ int ObMemtable::scan(
         }
       }
     } else {
-      ALLOCATE_TABLE_STORE_ROW_IETRATOR(context,
-            ObMemtableScanIterator,
-            scan_iter_ptr);
+      omt::ObTenantConfigGuard tenant_config(TENANT_CONF(MTL_ID()));
+      // TRANS_LOG(INFO, "Show tenant confg info", K(tenant_config.is_valid()), K(tenant_config));
+      // if (tenant_config.is_valid() && tenant_config->memtable_scan_filter_pushdown) {
+      //   ALLOCATE_TABLE_STORE_ROW_IETRATOR(context,
+      //         ObMemtableBlockScanIterator,
+      //         scan_iter_ptr);
+      //   TRANS_LOG(INFO, "Memtable scan iterator choice : blockscan");
+      // } else {
+      //   ALLOCATE_TABLE_STORE_ROW_IETRATOR(context,
+      //         ObMemtableScanIterator,
+      //         scan_iter_ptr);
+      //   TRANS_LOG(INFO, "Memtable scan iterator choice : normal");
+      // }
+      if (tenant_config.is_valid()) {
+        if (tenant_config->memtable_scan_filter_pushdown) {
```

Comment:
There are two suggestions:

1. The code here is a little redundant. 
2. We can use _pushdown_storage_level config